### DISCUSSION TITLE
[v6] Why is NavLink and Link not the same component?

### DISCUSSION THREAD

**Discussion by rwieruch** [remix-run/react-router#8275] [Q&A] [answered]
Is there a reason why there is an extra NavLink component just to access the `isActive` flag? I would think that in a perfect world, the Link component would just have this feature and there wouldn't be the need for another NavLink component.

**Comment by mjackson** [comment, marked as answer]
`<NavLink>` is really just a simple example of how to build a link that knows whether or not it is "active" by matching the beginning of the URL pathname. This is a common case in React Router because nested URL paths often correspond to nested UI.

There are other times when you might want a link to be active, such as when the URL contains a certain query parameter. For more complex cases, we provide the hooks for you to build your own `<NavLink>` as demonstrated in [this example](https://github.com/remix-run/react-router/blob/66bcb6c0f7123f511baecf85e20296dfb6584c4a/examples/custom-filter-link/src/App.tsx#L39-L59).
